app/controllers/instances_controller.rb

```diff
@@ -112,6 +125,12 @@ def assign_passcode(instance)
     instance
   end
 
+  def assign_qrcode(instance)
+    instance.qr_code = qr_code_url(passcode: instance.passcode)
```

Comment:
In the class they say we can put whatever url we want, and they actually use the official website of a restaurant (‘https://www.barravel.de’) However that’s not what we want, we need a link to our app, not a random external link. We cannot use a link_to or qr_code_path here because we need to save the whole url (qr_code_path only generate ‘/qr_code?passcode=blah’ so the url won’t be valid, it’s missing the ‘https://www.de-bored.fun’ part.) and also we cannot use interpolation and put back ‘https://www.de-bored.fun’ + qr_code_parh because  it won’t work when we run rails server on our local machine. So I figured the url helper was the best way :) I knew this helper before but can’t remember why! It was mentioned also in stackoverflow when I was checking how to pass a param in the url

Reply:
My goodness I can't find the stackoverflow anymore, but it was similar to the one below :) except more recent I believe
https://stackoverflow.com/questions/8531306/how-to-use-rails-named-route-helpers-with-parameters
![Screen Shot 2022-02-28 at 12 48 04](https://user-images.githubusercontent.com/68413600/155920876-103fde23-5a2e-4e19-9060-be59ec40a164.png)

Reply:
I think I know where its from now, click `url` under the **Helper column**. :D hiding from under out noses.
<img width="1256" alt="Screen Shot 2022-02-28 at 10 01 34" src="https://user-images.githubusercontent.com/18746187/155971174-b3d5aac0-32c6-485d-a008-a440f240001a.png">
.